Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,159 @@
+ * - Gyroscope (navx)
+ * - Encoders (grayhills)
+ * 
+ * Drivetrain methods:
+ * - arcadeDrive()
+ * - tankDrive()
+ * - reset()
+ * - getAngle()
+ */
+
+public class Drivetrain extends SubsystemBase {
+    /** CONTROLS */
+    private final DifferentialDriveKinematics driveKinematics;
+    private final Controller leftController;
+    private final Controller rightController;
+
+    private SmartNumber leftTargetSpeed;
+    private SmartNumber rightTargetSpeed;
+
+    /** HARDWARE */
+    private final MotorControllerGroup left;
+    private final MotorControllerGroup right;
+
+    private final Encoder leftGrayhill;
+    private final Encoder rightGrayhill;
+
+    private final AHRS navx;
+
+    /** ODOMETRY */
+    private final DifferentialDriveOdometry odometry;
+
+    private final Field2d field;
+
+    public Drivetrain() {
+        setSubsystem("Drivetrain");
+        /** CONTROL */
+        leftController = new PIDController(kP, kI, kD).add(new Feedforward.Drivetrain(kS, kV, kA).velocity());
+        rightController = new PIDController(kP, kI, kD).add(new Feedforward.Drivetrain(kS, kV, kA).velocity());
+
+        driveKinematics = new DifferentialDriveKinematics(Settings.Drivetrain.TRACK_WIDTH);
+
+        leftTargetSpeed = new SmartNumber("Drivetrain/Left Target Speed", 0.0);
+        rightTargetSpeed = new SmartNumber("Drivetrain/Right Target Speed", 0.0);
+
+        /** MOTORS */
+        CANSparkMax leftFront = new CANSparkMax(Ports.Drivetrain.LEFT_FRONT, MotorType.kBrushed);
+        CANSparkMax leftMiddle = new CANSparkMax(Ports.Drivetrain.LEFT_MIDDLE, MotorType.kBrushless);
+        CANSparkMax leftBottom = new CANSparkMax(Ports.Drivetrain.LEFT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.left.configure(leftFront, leftMiddle, leftBottom);
+        left = new MotorControllerGroup(leftFront, leftMiddle, leftBottom);
+        addChild("Left Motor Controller Group", left);
+
+        CANSparkMax rightFront = new CANSparkMax(Ports.Drivetrain.RIGHT_FRONT, MotorType.kBrushless);
+        CANSparkMax rightMiddle = new CANSparkMax(Ports.Drivetrain.RIGHT_MIDDLE, MotorType.kBrushless);
+        CANSparkMax rightBottom = new CANSparkMax(Ports.Drivetrain.RIGHT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.right.configure(rightFront, rightMiddle, rightBottom);
+        right = new MotorControllerGroup(rightFront, rightMiddle, rightBottom);
+        addChild("Right Motor Controller Group", right);
+
+        /** ENCODERS */
+        leftGrayhill = new Encoder(LEFT_A, LEFT_B);
+        rightGrayhill = new Encoder(RIGHT_A, RIGHT_B);
+        addChild("Left Encoder", leftGrayhill);
+        addChild("Right Encoder", rightGrayhill);
+
+        /** GYRO */
+        navx = new AHRS(SPI.Port.kMXP);
+        addChild("Gyroscope", navx);
+
+        /** ODOMETRY */
+        odometry = new DifferentialDriveOdometry(navx.getRotation2d());
+
+        field = new Field2d();
+        addChild("Field", field);
```

Comment:
Add drivetrain simulation

Reply:
yea -- idk if u should add child the field tho? just see if it works. i feel like u dont want the field in the drivetrain namespace